test_altimeter.ino

```diff
@@ -0,0 +1,45 @@
+#if false
+
+#include "altimeter.h"
+
+Altimeter alt;
+
+void setup() {
+  Serial.begin(9600);
+  while (!Serial) { /* Wait for Serial monitor to connect */ }
+  Serial.println("Connecting to sensor...");
+  alt.initialize();
+}
+
+void loop() {
+  // Exponential backoff: each time it waits longer than the last.
+  // That way, if a wire is loose or missing, it doesn't flood the console with messages.
+  static unsigned int backoff = 100;
+
+  if (alt.getStatus() != Altimeter::ACTIVE) {
+    Serial.print("Not connected. Trying again in ");
+    Serial.print(backoff);
+    Serial.println("ms...");
+    delay(backoff);
+    backoff *= 2;
+    alt.initialize();
+  } else {
+    float altitude;
+    setSeaLevel(altitude);
+
+    if (getAltitude() > m_seaLevel) {
```

Comment:
The `if` statement here isn't necessary. For the other sensors, the library asks the sensor if there is a value available, hence the `if (getValue(&value))` pattern. For the altimeter, it just waits until a value is available, so there's no need for that. You can just say `altitude = alt.getAltitude()`.

Reply:
Maybe the if statement here is unneeded, but I like the logic check of "hey is the altitude you gave me greater than sea level". So maybe get the altitude like Riley is saying, then perform a sanity check afterwards.

Reply:
I'm confused because you're saying `setSeaLevel()` returns altitude, but you're also saying that I should do `const float SEA_LEVEL_PRESSURE` and `setSeaLevel(SEA_LEVEL_PRESSURE)`. That doesn't make sense - can you clarify?

Reply:
`setSeaLevel` doesn't return anything -- its return type is `void`. The class has the following methods:

- `void initialize()`: establish I2C communications and, if possible, set the current air pressure to the sea level pressure.
- `void setSeaLevel(float seaLevel)`: set the sea level air pressure to the given number (in hPa). This number is necessary for the altitude calculations.
- `Status getStatus()`: check whether I2C is connected and whether the sea level is set.
- `float getAltitude()`: get the current altitude in feet.